Classes/common/query/CDTQQuerySqlTranslator.m

```diff
@@ -85,27 +85,20 @@ + (CDTQQueryNode *)translateQuery:(NSDictionary *)query
         return nil;
     } else if (!state.textIndexRequired &&
                   (!state.atLeastOneIndexUsed || state.atLeastOneORIndexMissing)) {
-        // If we haven't used a single index, we need to return a query
-        // which returns every document, so the posthoc matcher can
+        // If we haven't used a single index or an OR clause is missing an index,
+        // we need to return every document id, so that the post-hoc matcher can
         // run over every document to manually carry out the query.
+        CDTQSqlQueryNode *sqlNode = [[CDTQSqlQueryNode alloc] init];
         NSSet *neededFields = [NSSet setWithObject:@"_id"];
-        NSString *allDocsIndex =
-            [CDTQQuerySqlTranslator chooseIndexForFields:neededFields fromIndexes:indexes];
+        NSString *allDocsIndex = [CDTQQuerySqlTranslator chooseIndexForFields:neededFields
+                                                                  fromIndexes:indexes];
 
-        if (!allDocsIndex) {
-            LogError(@"No indexes defined, cannot execute query for all documents");
-            return nil;
+        if (allDocsIndex.length > 0) {
```

Comment:
In order for this fix to work, we populate the SQLNode with a SELECT statement ONLY if an index table exists.  If an index does not exist then we obviously cannot construct a SELECT statement.  Instead we leave the SQLNode alone and when the query node gets passed back to the query executor, the executor (https://github.com/cloudant/CDTDatastore/blob/45362-fix-create-default-index/Classes/common/query/CDTQQueryExecutor.m#L267) in turn will handle the "empty" SQL SELECT condition by getting the list of doc ids from the datastore.